api/api.go

```diff
@@ -124,9 +127,16 @@ func (api *API) Serve(ctx context.Context) error {
 	}()
 
 	logger.Info("starting server", "port", api.port)
-	if err := api.srv.ListenAndServe(); err != nil && err != http.ErrServerClosed {
-		logger.Error("error serving api", "port", api.port, "error", err)
-		return err
+	if config.BoolVal(api.tls.Enabled) {
+		if err := api.srv.ListenAndServeTLS(*api.tls.Cert, *api.tls.Key); err != nil && err != http.ErrServerClosed {
+			logger.Error("error serving api", "port", api.port, "error", err)
+			return err
+		}
+	} else {
+		if err := api.srv.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			logger.Error("error serving api", "port", api.port, "error", err)
+			return err
+		}
```

Comment:
```suggestion
	var err error
	if config.BoolVal(api.tls.Enabled) {
		err = api.srv.ListenAndServeTLS(*api.tls.Cert, *api.tls.Key)
	} else {
		err = api.srv.ListenAndServe()
	}
	if err != nil && err != http.ErrServerClosed {
		logger.Error("error serving api", "port", api.port, "error", err)
		return err
```

minor suggestion to reduce the nested ifs and identical error handling